### PR TITLE
Appication controller (probably wrong)

### PR DESCRIPTION
**Pull request by nikstojkov**

### CODE REVIEW

File: app/controllers/applications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @application.save
+    if @application.save
+      redirect_to opportunity_path(@opportunity)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @application = Application.find(params[:application_id])
```
Comment (JDS478) [inline app/controllers/applications_controller.rb:25]: Maybe add to the before_action instead?

**Review by Nick-ODonoghue — COMMENTED** [review]
Think before action should only be for show edit update and destroy actions.

Also def delete - Should this be def destroy?

File: app/controllers/applications_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class ApplicationsController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:show]
+  before_action :set_opportunity, only: [:new]
```
Comment (nikstojkov) [inline app/controllers/applications_controller.rb:3]: Add create

File: app/controllers/applications_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  def index
+    @applications = Application.all
+  end
+
+  def new
+    @application = Application.new
+  end
+
+  def create
```
Comment (nikstojkov) [inline app/controllers/applications_controller.rb:13]: Application.user = @user/current_user

File: app/controllers/applications_controller.rb (hunk 1 above)
Comment (nikstojkov) [inline app/controllers/applications_controller.rb:25]: Just params[:I'd]

File: app/controllers/applications_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @application = Application.find(params[:application_id])
+  end
+
+  def delete
+    @review.destroy
```
Comment (nikstojkov) [inline app/controllers/applications_controller.rb:29]: @application.destroy.... not review!!!

**Review by sulphite — CHANGES_REQUESTED** [review]

File: app/controllers/artworks_controller.rb [hunk 5]
```diff
@@ -11,7 +11,11 @@ def new
   def create
     artwork = Artwork.new(artwork_params)
     artwork.save
```
[the remark this answers is not in the document]
Comment (nikstojkov) [reply app/controllers/artworks_controller.rb:13]: im confused now. line 14?

[the remark this answers is not in the document]
Comment (sulphite) [reply app/controllers/artworks_controller.rb:13]: line 13, right after this you have `if artwork.save`

[the remark this answers is not in the document]
Comment (nikstojkov) [reply app/controllers/artworks_controller.rb:13]: Gotcha, i think in my head it was "try to save" -> "if saved do this"... There's the same code in the opportunities controller.

**Review by sulphite — APPROVED** [review]
looks good

**Pull request merged**